spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+
+=== Specify Config Profile
+
+The config profile can be specified via a system property or an environment viable `mp.config.profile`. It can be set when starting your microservice. The value of the property can contain only characters that are valid for config property names. 
+This is because the name of the profile is directly stored in the name of the config property. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The property `mp.config.profile` is only read once on microservice starting. Any changes after that is ignored. Furthermore, the value specifies a single profile. Only one profile can be active at a time.
```

Comment:
```suggestion
The property `mp.config.profile` shouldn't change after the JVM is started. It's only read once during JVM start up. If the property is modified afterwards the behavior is undefined and any changes to its value made later can be ignored by the implementation. Furthermore, the value specifies a single profile. Only one profile can be active at a time.
```

We shouldn't refer to "microservices" which is a vague term and not all MP implementations are used to build microservices. Rather I'd refer to a running JVM instead of a microservice.

Reply:
updated.

Reply:
I had  a further thought on this. I am going to change to application starting instead of JVM startup as it is more accurate to say application starting due to some application server might have multiple applications installed on the same server.

Reply:
How about on Config creation?

Reply:
> How about on Config creation?

This will be from implementation point of view though, which is different from user perspective. Some implementations might create the config a lot earlier.

Reply:
We can all agree that the profile is read one time.  But it doesn't make sense to specify exactly when it must be read; all we can really say for sure is that the implementation will (of a necessity) read the profile property before the application reads or injects configuration properties.  Since that's all the spec can reasonably require, that's exactly what should be specified.  Obviously this doesn't happen during JVM startup; it might not happen during application startup either.